AdyenActions/Actions/BoletoVoucherAction.swift

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+import Foundation
+
+/// Describes an action in which a Boleto voucher is presented to the shopper.
+public final class BoletoVoucherAction: Decodable {
+    
+    /// The `paymentMethodType` for which the voucher is presented.
+    public let paymentMethodType: VoucherPaymentMethod
+
+    /// The totalAmount  amount.
+    public let totalAmount: Payment.Amount
+
+    /// The payment reference.
+    public let reference: String
+
+    /// Expiry Date.
+    public let expiresAt: Date
+    
+    /// Download URL
+    public let downloadUrl: URL
+    
+    /// :nodoc:
+    public required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        
+        paymentMethodType = try container.decode(VoucherPaymentMethod.self, forKey: .paymentMethodType)
+        totalAmount = try container.decode(Payment.Amount.self, forKey: .totalAmount)
+        reference = try container.decode(String.self, forKey: .reference)
+        downloadUrl = try container.decode(URL.self, forKey: .downloadUrl)
+        
+        let expiresAtString = try container.decode(String.self, forKey: .expiresAt)
+        let dateFormatter = DateFormatter()
```

Comment:
We can use `ISO8601DateFormatter` right?

Reply:
yes, good find!

Reply:
Same for GenericVoucherAction, I believe

Reply:
yep, already fixed 👍

Reply:
BTW there is [duplication](https://sonarcloud.io/component_measures?id=Adyen_adyen-ios&metric=new_duplicated_lines_density&pullRequest=507&selected=Adyen_adyen-ios%3AAdyenActions%2FActions%2FBoletoVoucherAction.swift&view=list) with GenericVoucherAction. Maybe we can reuse it somehow 🤔

Reply:
yes. `BoletoVoucherAction` is a strange case of voucher action. it shares some fields, but not all, and adds some on top 🤷